tests/unit/src/flixel/input/mouse/FlxMouseEventManagerTest.hx

```diff
@@ -0,0 +1,54 @@
+package flixel.input.mouse;
+
+import flixel.input.mouse.FlxMouseEventManager;
+import massive.munit.Assert;
+
+class FlxMouseEventManagerTest
+{
+	@Before
+	function before()
+	{
+		FlxMouseEventManager.removeAll();
+	}
+	
+	@Test
+	@:access(flixel.input.mouse.FlxMouseEventManager)
+	function testMouseChildrenOrder()
+	{
+		var sprite0 = new FlxSprite();
+		var sprite1 = new FlxSprite();
+		var sprite2 = new FlxSprite();
+		var sprite3 = new FlxSprite();
+		
+		FlxMouseEventManager.add(sprite0, null, null, null, null, true);
+		FlxMouseEventManager.add(sprite1, null, null, null, null, false);
+		FlxMouseEventManager.add(sprite2, null, null, null, null, true);
+		FlxMouseEventManager.add(sprite3, null, null, null, null, false);
+		
+		// mouseChildren false comes before mouseChildren true, and more recently added comes before existing
+		var sprites = [sprite3, sprite1, sprite2, sprite0];
+		for (i in 0...sprites.length)
+		{
+			Assert.areEqual(sprites[i], FlxMouseEventManager._registeredObjects[i].object);
```

Comment:
This is really hacky, a unit test shouldn't need to rely on `@:access()`, only public API.

>the parameter set to false still allowed overlapping objects to receive events.

You should be able to test for that not happening anymore?

Reply:
(you should be able to dispatch mouse events manually, similar to what we do here:
https://github.com/HaxeFlixel/flixel/blob/dev/tests/unit/src/flixel/input/gamepad/FlxGamepadManagerTest.hx)

Reply:
Ohhh I didn't think of that. That's a much better test, yeah.